pandas/core/dtypes/cast.py

```diff
@@ -1026,3 +1027,52 @@ def find_common_type(types):
             return np.object
 
     return np.find_common_type(types, [])
+
+
+def maybe_cast_to_integer(arr, dtype):
+    """
+    Takes an integer dtype and returns the casted version, raising for an
+    incompatible dtype.
+
+    Parameters
+    ----------
+    arr : ndarray
+    dtype : np.dtype
+
+    Returns
+    -------
+    integer or unsigned integer array
+
+    Raises
+    ------
+    OverflowError
+        * If ``dtype`` is incompatible
+    ValueError
+        * If coercion from float to integer loses precision
+
+    Examples
+    --------
+    If you try to coerce negative values to unsigned integers, it raises:
+
+    >>> Series([-1], dtype='uint64')
+    Traceback (most recent call last):
+        ...
+    OverflowError: Trying to coerce negative values to unsigned integers
+
+    Also, if you try to coerce float values to integers, it raises:
+    >>> Series([1, 2, 3.5], dtype='int64')
+    Traceback (most recent call last):
+        ...
+    ValueError: Trying to coerce float values to integers
+
+    """
+
+    if is_unsigned_integer_dtype(dtype) and (arr < 0).any():
+        raise OverflowError("Trying to coerce negative values to unsigned "
+                            "integers")
+    elif is_integer_dtype(dtype) and (is_float_dtype(arr) or
+                                      is_object_dtype(arr)):
+        if not (arr == arr.astype(dtype)).all():
+            raise ValueError("Trying to coerce float values to integers")
+
+    return arr.astype(dtype, copy=False)
```

Comment:
I don't think you should cast here. This will coerce non-integer arrays to the passed dtype, which may not be nice.

instead branch on each of the if's (the integer and unsigned cases) and do an astype or raise).

so this will then explicity only work on integer dtypes that are passed, and not implicity on anything else